### PR TITLE
LPS-188301 and LPS-188303 Add friendly URL warnings when there is a conflict between a site template and some connected site

### PR DESCRIPTION
**Pull request by lfbesada**
https://liferay.atlassian.net/browse/LPS-188301
https://liferay.atlassian.net/browse/LPS-188303

cc @p2kmgcl ❤️

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by lfbesada** [comment]
ci:test:sf

**Comment by lfbesada** [comment]
ci:test:relevant

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/GetFriendlyURLWarningResourceCommand.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		LayoutSet layoutSet = _layoutSetLocalService.getLayoutSet(
+			group.getGroupId(), privateLayout);
+
+		if (group.isLayoutSetPrototype() ||
+			layoutSet.isLayoutSetPrototypeLinkEnabled()) {
+
+			return _language.get(
+				locale,
+				"the-friendly-url-of-the-site-template-page-you-are-trying-" +
+					"to-save-conflicts");
```
Comment (ruben-pulido) [inline modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/GetFriendlyURLWarningResourceCommand.java:180]: This sentence sounds "unfinished". I would have expected it to end with "-conflicts-with-x" or to have the form "x-and-x-conflict"

**Review by ruben-pulido — CHANGES_REQUESTED** [review]

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 2]
```diff
@@ -12573,6 +12573,7 @@ please-contact-the-administrator-to-assign-reviewers=Please contact the administ
 please-contact-the-administrator-to-configure-a-google-license=Please contact the administrator to configure a Google license.
 please-contact-the-administrator-to-configure-a-valid-mail-server=Please contact the administrator to configure a valid mail server.
 please-contact-the-administrator-to-configure-an-amazon-license=Please contact the administrator to configure an Amazon license.
+please-contact-the-administrator-to-resolve-this-friendly-url-conflict=Please contact the administrator to resolve This friendly URL conflict
```
[the remark this answers is not in the document]
Comment (ruben-pulido) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:12576]: "This" with an uppercase "T" also looks strange

[the remark this answers is not in the document]
Comment (lfbesada) [reply modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:12576]: This message is used between parenthesis which is the reason I didn't use a full stop at the end.
Yes, the uppercase "T" is very strange but "gw buildLang" change this "t" to "T" 🤷‍♀️

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutsAdminDisplayContext.java [hunk 3]
```diff
@@ -2138,6 +2230,53 @@ private long[] _getGroupIds() {
...
+
+			try {
+				String link = linkTag.doTagAsString(
+					httpServletRequest,
+					PortalUtil.getHttpServletResponse(_liferayPortletResponse));
+
+				Group group = layout.getGroup();
+
+				return LanguageUtil.format(
+					themeDisplay.getLocale(), "page-x-of-x",
```
[the remark this answers is not in the document]
Comment (lfbesada) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutsAdminDisplayContext.java:2252]: Agree

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutsAdminDisplayContext.java [hunk 4]
```diff
@@ -2138,6 +2230,53 @@ private long[] _getGroupIds() {
...
+				_log.error(jspException);
+			}
+		}
+
+		Group group = layout.getGroup();
+
+		return com.liferay.portal.kernel.util.StringUtil.
+			appendParentheticalSuffix(
+				LanguageUtil.format(
+					themeDisplay.getLocale(), "page-x-of-x",
```
[the remark this answers is not in the document]
Comment (lfbesada) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/display/context/LayoutsAdminDisplayContext.java:2268]: Agree

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/GetFriendlyURLWarningResourceCommand.java (hunk 1 above)
Comment (lfbesada) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/GetFriendlyURLWarningResourceCommand.java:180]: Agree

File: modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/GetFriendlyURLWarningResourceCommand.java [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+		}
+
+		return _language.get(
+			locale,
+			"the-friendly-url-of-the-page-you-are-trying-to-save-conflicts");
```
[the remark this answers is not in the document]
Comment (lfbesada) [reply modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/portlet/action/GetFriendlyURLWarningResourceCommand.java:185]: Totally agree

**Comment by ruben-pulido** [comment]
ci:reevaluate:1317401_8540

**Comment by lfbesada** [comment]
This failure [LocalFile.FriendlyURLService#CanRestoreOldFriendlyURL](https://test-1-26.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/674943//testReport/com.liferay.poshi.runner/PoshiRunner/test_LocalFile_FriendlyURLService_CanRestoreOldFriendlyURL_) is related to this PR, we are working to fix it.

**Comment by p2kmgcl** [comment]
Moved here: https://github.com/liferay-echo/liferay-portal/pull/12615

**Pull request closed without merging**